### PR TITLE
Dapa 66 router service separation [POST] /api/v1/git_tokens/ apply router service separation

### PR DESCRIPTION
**Pull request by mohamad-jaafar-monty**
feat: Added Service/Router separation for [POST] /api/v1/git_tokens/

Summary:

- Split the router into a router layer and a service layer.
- abstracted the database communication between the service layer and the Database Model.

## Summary by CodeRabbit

- **New Features**
  - Introduced a new service for adding and validating GitHub and GitLab tokens, including token encryption and validation with external APIs.
  - Improved error handling and standardized API responses when adding Git tokens.
  - Added a new request schema to streamline Git token payload handling.

- **Refactor**
  - Simplified the process for adding Git tokens by delegating logic to a dedicated service layer, reducing complexity in the API endpoint.

- **Chores**
  - Updated and consolidated token-related constants.
  - Removed tests for certain edge cases related to token labels and malformed UUIDs.

### CODE REVIEW

File: devdox/app/routes/git_tokens.py [hunk 1]
```diff
@@ -23,109 +23,17 @@
     GitLabelBase,
     GitLabelCreate,
 )
-from app.services.git_tokens_service import GetGitLabelService
+from app.services.git_tokens_service import GetGitLabelService, handle_github, handle_gitlab, PostGitLabelService
```
Comment (Hayat55) [inline devdox/app/routes/git_tokens.py]: remove unused functions like handle_github , handle_gitlab

**Review by Hayat55 — APPROVED** [review]

File: devdox/app/routes/git_tokens.py (hunk 1 above)
Comment (mohamad-jaafar-monty) [reply devdox/app/routes/git_tokens.py]: Done, resolved in [303fc6e](https://github.com/montymobile1/devdox-ai-portal-api/pull/61/commits/303fc6ecc3dd29488696a33cc8213dedaaf6d73e)

**Review by Hayat55 — APPROVED** [review]

File: devdox/app/services/git_tokens_service.py [hunk 2]
```diff
@@ -87,3 +105,144 @@ async def get_git_labels_by_label(self, pagination: PaginationParams, user_claim
...
+
+        return APIResponse.success(
+            message=constants.TOKEN_SAVED_SUCCESSFULLY, data={"id": str(git_label.id)}
+        )
+    except Exception:
+        logger.exception(
+            "Unexpected Failure while attempting to save GitHub token on Path = '[POST] /api/v1/git_tokens' -> handle_github"
+        )
+        return APIResponse.error(message=constants.GITHUB_TOKEN_SAVE_FAILED)
+
```
[the remark this answers is not in the document]
Comment (mohamad-jaafar-monty) [reply devdox/app/services/git_tokens_service.py:197]: I am aware, this code is being uprooted, it has been fully replaced in https://github.com/montymobile1/devdox-ai-portal-api/pull/64

File: devdox/app/services/git_tokens_service.py [hunk 3]
```diff
@@ -87,3 +105,144 @@ async def get_git_labels_by_label(self, pagination: PaginationParams, user_claim
...
+    except Exception:
+        logger.exception(
+            "Unexpected Failure while attempting to save GitLab token on Path = '[POST] /api/v1/git_tokens' -> handle_gitlab"
+        )
+
+        return APIResponse.error(
+            message=SERVICE_UNAVAILABLE,
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+        )
+
```
[the remark this answers is not in the document]
Comment (mohamad-jaafar-monty) [reply devdox/app/services/git_tokens_service.py:166]: I am aware, this code is being uprooted, it has been fully replaced in https://github.com/montymobile1/devdox-ai-portal-api/pull/64

File: devdox/app/routes/git_tokens.py [hunk 4]
```diff
@@ -205,59 +108,21 @@ async def get_git_label_by_label(
...
+        json_payload=payload
+    )
 
-        return APIResponse.error(
-            message=SERVICE_UNAVAILABLE,
-            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
-        )
+    return APIResponse.success(
+        message=constants.TOKEN_SAVED_SUCCESSFULLY, data={"id": str(results.id)}
+    )
```
[the remark this answers is not in the document]
Comment (mohamad-jaafar-monty) [reply devdox/app/routes/git_tokens.py]: This is a router, an API, it's supposed to return that, and as explained earlier, the service layer has been fully addressed and integrated in https://github.com/montymobile1/devdox-ai-portal-api/pull/64 with proper return, this PR is only meant to split the router into a service <--> router
